src/plots/polar/set_convert.js

```diff
@@ -0,0 +1,169 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var setConvertCartesian = require('../cartesian/set_convert');
+
+var deg2rad = Lib.deg2rad;
+var rad2deg = Lib.rad2deg;
+var isFullCircle = Lib.isFullCircle;
+
+/**
+ * setConvert for polar axes!
+ *
+ * @param {object} ax
+ *   axis in question (works for both radial and angular axes)
+ * @param {object} polarLayout
+ *   full polar layout of the subplot associated with 'ax'
+ * @param {object} fullLayout
+ *   full layout
+ *
+ * Here, reuse some of the Cartesian setConvert logic,
+ * but we must extend some of it, as both radial and angular axes
+ * don't have domains and angular axes don't have _true_ ranges.
+ *
+ * Moreover, we introduce two new coordinate systems:
+ * - 'g' for geometric coordinates and
+ * - 't' for angular ticks
+ *
+ * Radial axis coordinate systems flow:
+ * - d2c (in calc) just like for cartesian axes
+ * - c2g (in plot) translates calcdata about `radialaxis.range[0]`
+ *
+ * Angular axis coordinate systems flow:
+ * + for linear axes:
+ *   - d2c (in calc) angles -> 'data' radians
+ *   - c2g (in plot) 'data' -> 'geometric' radians (fn of ax rotation & direction)
+ *   - t2g (in updateAngularAxis) 'tick' value (in degrees) -> 'geometric' radians
+ * + for category axes:
+ *   - d2c (in calc) just like for cartesian axes
+ *   - c2g (in plot) category indices -> 'geometric' radians
+ *   - t2g (in updateAngularAxis) 'tick' value (as category indices) -> 'geometric' radians
```

Comment:
This would be easier for me to understand if instead of describing `t2g` etc, just describe `t` and `g` (and `c` and `d` if they're different from cartesian) for the linear and category cases)

Reply:
(like https://github.com/plotly/plotly.js/blob/def6aa5a24527c68e90f50485255170609640e43/src/plots/cartesian/set_convert.js#L35-L50)

Reply:
latest attempt in -> 4c11c84251cfe8aa532f42a26022803e867f4ed4

Reply:
Much clearer - thanks!